massa-execution-worker/src/speculative_async_pool.rs

```diff
@@ -287,7 +287,7 @@ impl SpeculativeAsyncPool {
                 }
                 Present(SetUpdateOrDelete::Update(msg_update)) => {
                     current_changes.entry(message_id).and_modify(|e| {
-                        e.apply(msg_update.clone());
+                        *e = msg_update.clone();
```

Comment:
why was this changed ?

Reply:
It did work before, but it technically applied the current_changes twice (the returned msg_update already took them into account, so we directly replace it)